lib/rule-list.js

```diff
@@ -155,17 +155,187 @@ export const SIMPLIFY_FRACTION = defineRuleString(
+
+    // simplified the fraction if possible 
+    const simplified = canApplyRule(SIMPLIFY_FRACTION, fraction)
+          ? applyRule(SIMPLIFY_FRACTION, fraction)
+          : fraction
+
+    const newNumerator = parseInt(int) * query.getValue(simplified.args[1]) + query.getValue(simplified.args[0])
+    return build.div(build.number(newNumerator), simplified.args[1])
+}
+
+// e.g. 2 + 3/2 -> 2/1 + 3/2
+// e.g. 1.2 + 3/2 -> 6/5 + 3/2
+export const CONVERT_TO_FRACTION = defineRule(
+    (node) => {
+        if (query.isAdd(node)) {
+            const terms = node.args
+            // only some of the terms are fractions
+            const hasFraction = terms.some(term => isFraction(term)) && !terms.every(term => isFraction(term))
+
+            return hasFraction ? {node} : null
+        }
+    },
+
+    (node) => {
+        let terms = node.args
+
+        const negatives = getNegatives(terms)
+
+        terms = terms.map(function(term) {
+            let newTerm
+            if (isDecimal(term)) {
+                // 2.2 and -2.2
+                newTerm = query.isNeg(term)
+                    ? decimal_to_fraction(term.args[0])
+                    : decimal_to_fraction(term)
+            } else if (isPolynomialTerm(term)) {
+                newTerm = query.isNeg(term)
+                ? build.div(term.args[0], build.number(1))
+                : build.div(term, build.number(1))
+            } else {
+                newTerm = query.isNeg(term)
+                    ? term.args[0]
+                    : term
+            }
+            return newTerm
+        })
+
+        const result = build.add(
+            ...terms.map((term, i) => negatives[i]
+                         ? build.neg(term, {wasMinus: true})
+                         : term))
+        return result
+    }
+)
+
+// Finds common denominator in any scenario
+
+// e.g. 2/6 + 1/4 -> (2 * 2) / (6 * 2) + (1 * 3) / (4 * 3)
+// e.g. 2/(3 + x) + 2/3 -> (2 * 3)/ (3 * (3 + x)) + 2 * (3 + x) / (3 * (3 + x))
+// e.g. 3 + 2/3 -> (3 * 3) / 3 + 2 / 3
+export const COMMON_DENOMINATOR = defineRule(
+    (node) => {
+        if (query.isAdd(node)) {
+            const terms = node.args
+            
+            // checks if at least one of the terms is a fraction
+            const hasFraction = terms.some(term => isFraction(term))
+ 
+            let sameDenom
+            // rule cannot apply if all of the terms are fractions
+            // and they all have the same denominator
+            if (terms.every(term => isFraction(term))) {
+                const denom = getDenominator(terms[0])
+
+                sameDenom = terms.every(term =>
+                                        print(getDenominator(term)) == print(denom))
+            }
+
+            return hasFraction && !sameDenom ? {node} : null
+        }
+    },
+
+    (node) => {
+        node = canApplyRule(CONVERT_TO_FRACTION, node)
+            ? applyRule(CONVERT_TO_FRACTION, node)
+            : node
```

Comment:
I see, it has to have at least one fraction... I obviously need to get more sleep.  :P